src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationExtensions.cs

```diff
@@ -113,6 +113,7 @@ internal static CSharpCompilation ToCompilation(this ImmutableArray<MetadataBloc
             platform: Platform.AnyCpu, // Platform should match PEModule.Machine, in this case I386.
             optimizationLevel: OptimizationLevel.Release,
             assemblyIdentityComparer: DesktopAssemblyIdentityComparer.Default).
-            WithMetadataImportOptions(MetadataImportOptions.All);
+            WithMetadataImportOptions(MetadataImportOptions.All).
+            WithModuleName("1");
```

Comment:
@tmat Please tell me if this is OK. It was required to get tests to pass, as otherwise we use a random GUID for the module name in E&C, which means that the generated code is not deterministic even for testing.